### ISSUE TITLE
Searching by non-existent name should not 404

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#647]
Currently when we run a search with a name specified which does not exist, the server returns a NOT FOUND error. For example,

```
$ python client_dev.py readgroupsets-search http://localhost:8000 --name=SDF
ERROR:ga4gh.client:404 {"errorCode": 983484777, "message": "ReadGroupSet with name 'SDF' not found"}
```

This should return an empty set of results. I think ``readGroupSetGenerator`` and ``callSetGenerator`` are affected by this in ``backend.py``.

@hjellinek, what do the compliance tests expect here?

**Comment by dcolligan** [comment]
That's not the error I get for that command on develop:
```
Traceback (most recent call last):
  File "/Users/dannyc/ga4gh/new/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/dannyc/ga4gh/new/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/dannyc/ga4gh/server/ga4gh/frontend.py", line 461, in searchReadGroupSets
    flask.request, app.backend.runSearchReadGroupSets)
  File "/Users/dannyc/ga4gh/server/ga4gh/frontend.py", line 391, in handleFlaskPostRequest
    return handleHttpPost(flaskRequest, endpoint)
  File "/Users/dannyc/ga4gh/server/ga4gh/frontend.py", line 268, in handleHttpPost
    responseStr = endpoint(request.get_data())
  File "/Users/dannyc/ga4gh/server/ga4gh/backend.py", line 681, in runSearchReadGroupSets
    self.readGroupSetsGenerator)
  File "/Users/dannyc/ga4gh/server/ga4gh/backend.py", line 556, in runSearchRequest
    for obj, nextPageToken in objectGenerator(request):
  File "/Users/dannyc/ga4gh/server/ga4gh/backend.py", line 405, in readGroupSetsGenerator
    dataset = self.getDataset(request.datasetId)
  File "/Users/dannyc/ga4gh/server/ga4gh/backend.py", line 271, in getDataset
    raise exceptions.DatasetNotFoundException(id_)
DatasetNotFoundException: The requested dataset 'None' was not found
```
Isn't our usual MO to return a 404 if any part of the query doesn't exist?  Why would we do things differently here?

**Comment by jeromekelleher** [comment]
Sorry, that command was run against #643 --- I'm really sick of copying and pasting all these IDs so I used that.

For your question, there is a subtle distinction here. If ``datasetId`` is set to something that doesn't exist, we should 404, because this is an explicit reference to something that isn't there. For ``name`` though, were saying something slightly different: this is a *filter* on the ReadGroupSets and therefore results in an empty set of results if nothing matches.

**Comment by hjellinek** [comment]
Sorry, I just noticed your question for me here.
The test `ReadGroupSetsGetByNameIT.testSearchForBogusNameReturnsEmptyList` actually does what its name implies": the `SearchReadGroupSetsResponse` for a nonexistent name is expected to contain an empty `readGroupSets` list.

**Comment by macieksmuga** [comment]
Thanks @hjellinek! 
+1 from me, then.

**Comment by dcolligan** [comment]
Fixed in #658 and #650
